src/diagram/components/quantity-node.tsx

```diff
@@ -201,26 +220,22 @@ const _QuantityNode: React.FC<IProps> = ({ data, isConnectable }) => {
           />
         </div>
         {hasExpression &&
-          <div className={classNames("variable-info-row", "expression-row", {"expanded": showExpression })} data-testid="variable-expression-row">
-            <textarea
-              autoComplete="off"
-              spellCheck="false"
-              className={classNames("variable-expression-area", { "invalid": variable.computedValueError || variable.computedUnitError })}
-              data-testid={"variable-expression"}
-              maxLength={kMaxNotesCharacters}
-              onChange={onExpressionChange}
-              placeholder={"expression"}
+          <div className={classNames("variable-info-row", "expression-row", {"expanded": (hasLongExpression && showFullExpression) })} data-testid="variable-expression-row">
+            <ExpandableInput
+              error={!!(variable.computedValueError || variable.computedUnitError)}
+              inputType="text"
+              lengthToExpand={kExpressionExpandLength}
+              maxLength={kMaxNameCharacters}
```

Comment:
Not sure what this should be but it needs to be longer. Maybe if `maxLength` is undefined there is no limit?